### ISSUE TITLE
appAccountToken not returned in purchase of purchaseUpdatedListener

### ISSUE THREAD

**Issue by CyberTron88** [dooboolab-community/react-native-iap#2752]
I passed in a uuid to the appAccountToken, this is not returned anywhere inside the purchase of purchaseUpdatedListener. The field is definitely passed in correctly as I see the value inside the notification received under transactionInfo.

I'm passing in as  

await requestSubscription({ sku: subscriptionSKU, appAccountToken: user_uuid });

**Comment by CyberTron88** [comment]
Ok. I found it inside latest_receipt_info but not in_app inside transactionReceipt.recipe

**Comment by Malhar62** [comment]
hi @CyberTron88 

can you please provide full explaination of it, I tried passing userID to appAccountToken but didn't get it in purchaseUpdatedListener or latest_receipt_info.

thanks for the help.

**Comment by fatidian1** [comment]
@Malhar62 It is working properly if you turn on storykit2 mode:
```js
    if (Platform.OS === 'ios') {
      storekit2Mode();
    }
    initConnection().then(() => {
    // ...
    });
```

**Comment by Malhar62** [comment]
I found the way, appAccountToken has to be in UUID format then only we will receive it from webhook notification in backend. Not in any random string.

**Comment by fatidian1** [comment]
Yeah, that's the first thing that I did, but in storykit1 mode it wasn't working.

**Comment by Malhar62** [comment]
Oh okay, I am using storekit2, Thank you.

**Comment by Karthik-V26** [comment]
Have you found any solution for this @Malhar62 ?

**Comment by Malhar62** [comment]
> @Karthik-V26 Can you show me your code? What kind of string is your appAccountToken? What platform are you testing it on?

@Karthik-V26 the type has to be UUID, apple has mentioned that on official docs otherwise it won't send token if it's in any other form.

**Comment by Karthik-V26** [comment]
@Malhar62 , You are right, If the type is UUID then I can see it. Thanks for the answer.
Could you help me in knowing the advantages of UUID, I actually need some id to send to backend to identify the user. 
Like to differntiate the two users who tries to buy subscription using same appleID to avoid conflicts?

**Comment by Malhar62** [comment]
> @Malhar62 , You are right, If the type is UUID then I can see it. Thanks for the answer. Could you help me in knowing the advantages of UUID, I actually need some id to send to backend to identify the user. Like to differntiate the two app account users who tries to buy subscription using same appleID to avoid conflicts? 

On backend you can attach UUID to a user so each will have their unique UUID key, so after purchase backend will get appaccountToken as UUID and can differentiate user by that.

**Comment by hyochan** [comment]
I’m closing all issues reported in versions below 14, as the library now supports the new architecture with NitroModules and has been completely revamped.
